presto-main/src/main/java/com/facebook/presto/operator/aggregation/MergeStatisticalDigestFunction.java

```diff
@@ -0,0 +1,153 @@
+import com.facebook.presto.operator.aggregation.state.StatisticalDigestStateSerializer;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.Signature;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.spi.type.TypeSignatureParameter;
+import com.facebook.presto.type.StatisticalDigest;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+import java.util.List;
+
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.AccumulatorStateDescriptor;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INDEX;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INPUT_CHANNEL;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
+import static com.facebook.presto.spi.function.FunctionKind.AGGREGATE;
+import static com.facebook.presto.spi.function.Signature.comparableTypeParameter;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static java.lang.String.format;
+
+public abstract class MergeStatisticalDigestFunction
+        extends SqlAggregationFunction
+{
+    public final String name;
+    public final String type;
+    public final StatisticalDigestStateFactory factory;
+
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(
+            MergeStatisticalDigestFunction.class,
+            "combine",
+            StatisticalDigestState.class,
+            StatisticalDigestState.class);
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(
+            MergeStatisticalDigestFunction.class,
+            "output",
+            StatisticalDigestStateSerializer.class,
+            StatisticalDigestState.class,
+            BlockBuilder.class);
+
+    MergeStatisticalDigestFunction(String name, String type, StatisticalDigestStateFactory factory, boolean hidden)
+    {
+        super(new Signature(
+                        name,
+                        AGGREGATE,
+                        ImmutableList.of(comparableTypeParameter("T")),
+                        ImmutableList.of(),
+                        parseTypeSignature(type + "(T)"),
+                        ImmutableList.of(parseTypeSignature(type + "(T)")),
+                        false),
+                hidden);
+        this.name = name;
+        this.type = type;
+        this.factory = factory;
+    }
+
+    @Override
+    public String getDescription()
```

Comment:
It's a strange concept for an abstract class to provide description. Even though in this case it works. I'd prefer having the final classes to override this.